### PR TITLE
feat(postgres): add postgres setup docker image

### PR DESCRIPTION
**Pull request by jensenity**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

## Description
Hi! This is my first PR here! Currently there's only a mysql-setup docker image on DockerHub. If we want to use postgres instead of mysql, we will need to manually add the table from [init.sql](https://github.com/linkedin/datahub/blob/master/docker/postgres/init.sql) to the postgres database. With this image, we can automate it by running this image, instead of manually running the sql file. It will also help deploying datahub on kubernetes by switching mysql setup image to postgres setup image and add the necessary environment variables to it. I know there isn't any official issue on this but there've been a few requests from the community slack. Hope i'm doing this right! 🙇‍♂️

### CODE REVIEW

**Comment by shirshanka** [comment]
Thanks for this PR @jensenity! 
We are considering simplifying the setup to not require separate setup images and having the main gms container auto-setup the storage + indexing layer. 
Will get back to you on whether we can make that happen quickly or if we should just create another setup container for now.

File: docker/postgres-setup/init.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
+-- create datahub database
+CREATE DATABASE IF NOT EXISTS DATAHUB_DB_NAME;
+USE DATAHUB_DB_NAME;
+
+-- create metadata aspect table
+CREATE TABLE metadata_aspect_v2 (
```
Comment (dexter-mh-lee) [inline docker/postgres-setup/init.sql:6]: Can we make this CREATE TABLE IF NOT EXISTS ?

**Review by dexter-mh-lee — COMMENTED** [review]
Thanks for the contribution!! Minor comment

**Comment by shirshanka** [comment]
Checked with @dexter-mh-lee, auto-setup will take a bit longer. So we'll take this PR in once you address the minor comments!

> Thanks for this PR @jensenity!
> We are considering simplifying the setup to not require separate setup images and having the main gms container auto-setup the storage + indexing layer.
> Will get back to you on whether we can make that happen quickly or if we should just create another setup container for now.

File: docker/postgres-setup/init.sql (hunk 1 above)
Comment (jensenity) [reply docker/postgres-setup/init.sql]: eecaf71

File: docker/postgres-setup/init.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  version                       bigint not null,
+  metadata                      text not null,
+  systemmetadata                text,
+  createdon                     timestamp not null,
+  createdby                     varchar(255) not null,
+  createdfor                    varchar(255),
+  CONSTRAINT pk_metadata_aspect_v2 PRIMARY KEY (urn, aspect, version)
+);
+
+INSERT INTO DATAHUB_DB_NAME.metadata_aspect_v2 (urn, aspect, version, metadata, createdon, createdby) VALUES(
```
Comment (dexter-mh-lee) [inline docker/postgres-setup/init.sql:17]: Can you follow what is done here? https://github.com/linkedin/datahub/blob/master/docker/mysql-setup/init.sql#L19 Ensures that duplicate rows are never added. You can ignore the metadata_index creation.

Comment (jensenity) [reply docker/postgres-setup/init.sql]: ad31985

just added the insert if not exists query^^

**Review by dexter-mh-lee — APPROVED** [review]
LGTM!! Thanks for the contribution!!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**